platform/docs/docs/platform/services/ui/ui-customization-service.md

```diff
@@ -0,0 +1,196 @@
+---
+# UI Customization Service
+
+There are a lot of places where users may want to configure certain elements
+differently between different modes or for different deployments.  A mode
+example might be the ability to select a different right click context menu
+for the cornerstone extension.  A deployment example might be the ability to add
+a selector for the data source back end to use to the study results table.
+
+The use of this service enables these to be defined in a typed fashion by
+providing an easy way to set default values for this, but to allow a non-default value to be specified by the configuration or mode.
+
+This service is a UI service in that part of the registration allows for registering
+UI components and types to deal with, but it does not directly provide an UI
+displayable elements unless customized to do so.
+
+## Registering Customizations
+There are several ways to register customizations.  The `APP_CONFIG.whiteLabeling`
+field is used as a per-configuration entry.  This object can list single
+configurations by id, or it can list sets of customizations by referring to
+the `customizationModule` in an extension.
+
+As well, extensions can register default customizations by provided a 'default'
+name key within the extension.
+
+Mode specific customizations can be registered in the `onModeEnter` of the mode.
+Note that mode specific customizations are CLEARED before this call is made
+by a call to the `uiCustomizationService.onModeEnter` from the `ExtensionManager`.
+
+The following example shows first the registration of the default values,
+in this case they are base types which are inheritted later:
+
+```js
+// In the cornerstone extension  getCustomizationModule:
+const getCustomizationModule = () => ([
+  {
+    name: 'default',
+    value: [
+      {
+        id: 'ohif.cornerstoneContextMenu',
+        commands: [... default commands for the context menu],
+        content: CornerstoneOverlay,
+        contentProps: ....
+      },
+      {
+        id: 'ohif.menuItem',
+        content: CornerstoneMenuItem,
+      },
+    ],
+  },
+]);
+```
+
+Then, in the configuration file one might have a custom overlay definition:
+
+```js
+// in the APP_CONFIG file set the top right area to show the patient name
+// using PN: as a prefix when the study has a non-empty patient name.
+whiteLabeling: {
+  cornerstoneOverlayTopRight: {
+    id: 'cornerstoneOverlayTopRight',
+    items: [
+      {
+        id: 'PatientNameOverlay',
+        // Note the overlayItem as a parent type - this provides the
+        // rendering functionality to read the attribute and use the label.
+        customizationType: 'ohif.overlayItem',
+        attribute: 'PatientName',
+        label: 'PN:',
+      },
+    ],
+  }
+},
+```
+
+In the mode customization, the context menu is then customized:
+
+```js
+// First in ExtensionManager, the mode customizations are cleared by calling:
+UICustomizationService.onModeEnter();
```

Comment:
This is a redundant information IMO, user does not need to know that this is getting called in the extension manager. They only need to know that the following code will run on a mode enter